pkg/logs/input/journald/scanner.go

```diff
@@ -0,0 +1,82 @@
+package journald
+
+import (
+	"strings"
+
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/auditor"
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/pipeline"
+	"github.com/DataDog/datadog-agent/pkg/logs/restart"
+)
+
+// Scanner is in charge of starting and stopping new journald tailers
+type Scanner struct {
+	sources          []*config.LogSource
+	pipelineProvider pipeline.Provider
+	auditor          *auditor.Auditor
+	tailers          map[string]*Tailer
+}
+
+// New returns a new Scanner.
+func New(sources []*config.LogSource, pipelineProvider pipeline.Provider, auditor *auditor.Auditor) *Scanner {
```

Comment:
I don't think the scanner does much logic that is systemd related, would it make sense to share more logic with other sources? in this case it doesn't scan, it just starts the tailers